server/stream.go

```diff
@@ -5513,15 +5516,30 @@ func (mset *stream) hasAllPreAcks(seq uint64, subj string) bool {
 // Check if we have all consumers pre-acked.
 // Write lock should be held.
 func (mset *stream) clearAllPreAcks(seq uint64) {
-	delete(mset.preAcks, seq)
+	for o, ss := range mset.preAcks {
```

Comment:
This is now O(n) vs O(1) depending on number of consumers. The interest policy usually has lower number of consumers but the general trend we have been observing is more and more consumers.

Reply:
Yes, not clear exactly what the right balance is here. Certainly a lot of memory is saved which prevents OOMs but does risk being marginally slower.

Reply:
Yes, but again preAck state was meant to be small. So better way to fix the exploding state space I think.